catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/catalog/CatalogApplication.java

```diff
@@ -0,0 +1,389 @@
+import org.apache.cxf.jaxrs.impl.UriBuilderImpl;
+import org.apache.http.HttpStatus;
+import org.codice.ddf.rest.service.CatalogException;
+import org.codice.ddf.rest.service.CatalogService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+import spark.servlet.SparkApplication;
+
+public class CatalogApplication implements SparkApplication {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CatalogApplication.class);
+
+  private static final String ECLIPSE_MULTIPART_CONFIG = "org.eclipse.jetty.multipartConfig";
+
+  private static final String JAVA_IO_TMPDIR = "java.io.tmpdir";
+
+  private CatalogService catalogService;
+
+  public CatalogApplication(CatalogService catalogService) {
+    this.catalogService = catalogService;
+  }
+
+  @Override
+  public void init() {
+    head(
+        "/catalog/",
+        (req, res) -> {
+          res.status(HttpStatus.SC_OK);
+          return res;
+        });
+
+    head("/catalog/:id", (req, res) -> getHeaders(req, res, null, req.params(":id")));
+
+    head(
+        "/catalog/sources/:sourceid/:id",
+        (req, res) -> getHeaders(req, res, req.params(":sourceid"), req.params(":id")));
+
+    get(
+        "/catalog/sources",
+        (req, res) -> {
+          final BinaryContent content = catalogService.getDocument();
+
+          res.status(HttpStatus.SC_OK);
+          res.body(IOUtils.toString(content.getInputStream(), StandardCharsets.UTF_8));
+          res.type(content.getMimeTypeValue());
+
+          // Add the Accept-ranges header to let the client know that we accept ranges in bytes
+          res.header(HEADER_ACCEPT_RANGES, BYTES);
+          return res;
+        });
+
+    get(
+        "/catalog/:id",
+        (req, res) -> getDocument(req, res, null, req.params(":id"), req.queryParams("transform")));
+
+    get(
+        "/catalog/sources/:sourceid/:id",
+        (req, res) ->
+            getDocument(
+                req,
+                res,
+                req.params(":sourceid"),
+                req.params(":id"),
+                req.queryParams("transform")));
+
+    post(
+        "/catalog/metacard",
+        (req, res) -> {
+          try {
+            final BinaryContent content =
+                catalogService.createMetacard(req.raw(), req.queryParams("transform"));
+
+            res.status(HttpStatus.SC_OK);
+            res.body(IOUtils.toString(content.getInputStream(), StandardCharsets.UTF_8));
+            res.type(content.getMimeTypeValue());
+            return res;
+          } catch (CatalogException e) {
+            return createBadRequestResponse(res, e.getMessage());
+          }
+        });
+
+    post(
+        "/catalog/",
+        (req, res) -> {
+          if (req.contentType().startsWith("multipart/")) {
+            req.attribute(
+                ECLIPSE_MULTIPART_CONFIG,
+                new MultipartConfigElement(System.getProperty(JAVA_IO_TMPDIR)));
```

Comment:
❓ What is this doing?

Reply:
You need to set a `MultipartConfigElement` to be able to work with multipart data.